src/main/sensors/gyro.c

```diff
@@ -44,20 +45,29 @@ float gyroADCf[XYZ_AXIS_COUNT];
 static int32_t gyroZero[XYZ_AXIS_COUNT] = { 0, 0, 0 };
 static const gyroConfig_t *gyroConfig;
 static biquadFilter_t gyroFilter[XYZ_AXIS_COUNT];
+static biquadFilter_t gyroFilterNotch[XYZ_AXIS_COUNT];
+static biquadFilter_t gyroFilterOld[XYZ_AXIS_COUNT];
```

Comment:
Yes biquad can easily be transformed for  different type of filters easily.

IMO this feature would not be user friendly. It means that on each
multirotor you would have to run frequency spectrum to determine a motor
noise frequency. Ok if user knows how to do it can be done.

Next problem.....oops different throttle levels / rpm need different notch
filter frequencies.

Yet another problem....different props need another notch frequency.

In other words definately not something to be happy with on dynamic
environment we have on multirotors. All you need is a good lowpass and
thats it.

If your setup still has heavy noise with all available lowpass options.
Gyro_lpf and soft lowpass than I am sorry to say your setup has a serious
issue and you should look at that and get some better equipment
Op 4 jul. 2016 10:23 a.m. schreef "rav-rav" <notifications@github.com>:

> In src/main/sensors/gyro.c
> <https://github.com/betaflight/betaflight/pull/668#discussion_r69423445>:
>
> > @@ -44,20 +45,29 @@ float gyroADCf[XYZ_AXIS_COUNT];
> >  static int32_t gyroZero[XYZ_AXIS_COUNT] = { 0, 0, 0 };
> >  static const gyroConfig_t *gyroConfig;
> >  static biquadFilter_t gyroFilter[XYZ_AXIS_COUNT];
> > +static biquadFilter_t gyroFilterNotch[XYZ_AXIS_COUNT];
> > +static biquadFilter_t gyroFilterOld[XYZ_AXIS_COUNT];
>
> You are right, I added gyroFilterOld for debugging purposes in order to
> directly compare the old and the new filter. It won't be necessary for
> normal flight.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/betaflight/betaflight/pull/668/files/dbb236e9df4fb6765a0a97bf5ce26b3b468818db#r69423445>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKQlhFbBj7pKbfzNuKnFx73UrCvcg383ks5qSMLpgaJpZM4JEEY0>
> .
>